internal/app/storage/storage.go

```diff
@@ -0,0 +1,25 @@
+package storage
+
+import (
+	"errors"
+)
+
+type Storage map[string]string
+
+func NewStorage() Storage {
+	arr := make(Storage)
+	return arr
+}
+
+func (strg *Storage) Save(key, value string) (string, error) {
+	(*strg)[key] = value
+	return key, nil
+}
+
+func (strg Storage) OldUrl(key string) (string, error) {
```

Comment:
Не копируем, а берем ссылку на текущий сторадж
```suggestion
func (s *Storage) OldUrl(key string) (string, error) {
```